azext_iot/central/providers/device_provider.py

```diff
@@ -246,3 +262,28 @@ def get_all_registration_info(
         ]
 
         return result
+
+    def get_registration_summary(self, central_dns_suffix="azureiotcentral.com"):
+        registration_summary = {}
+
+        devices = self.list_devices(central_dns_suffix=central_dns_suffix)
```

Comment:
lets move this to an object. Create something like

```
class RegistrationSummary():
  def __init__(self):
    self.counter = {name: 0 for name in DeviceStatus}
  
  def process_device(self, device: Device):
    self.counter[device.device_status.name] += 1
```

Then when calling, this code can be simplified to:
```
summary = RegistrationSummary()

devices = ...
[summary.process_device(device) for device in devices if not devices.simulated]
return summary.counter
```

Reply:
I'd highly suggest familiarizing yourself with list and dictionary comprehensions. They're really really important in python, and are faster than for-loops in general. More than happy to help if you have any questions about it